otter/log/formatters.py

```diff
@@ -215,6 +219,42 @@ def audit_log_formatter(eventDict, timestamp, hostname):
     return audit_log_params
 
 
+def ErrorFormattingWrapper(observer):
+    """
+    Return log observer that will format error if any and delegate it to
+    given `observer`
+    """
+
+    def error_formatting_observer(event):
+
+        message = ""
+
+        if event.get("isError", False):
+            level = 3
+
+            if 'failure' in event:
+                excp = event['failure'].value
+                message = repr(excp)
+                event['traceback'] = event['failure'].getTraceback()
+                event['exception_type'] = excp.__class__.__name__
+
+            if 'why' in event and event['why']:
+                message = '{0}: {1}'.format(event['why'], message)
+
+        else:
+            level = 6
+
+        event.update({
+            "message": event.get("message", (message, )),
```

Comment:
Oh ok.  I'm not sure how to handle that.  :|  But it seems like otherwise the message gets emptied out:

Current traceback messages look like:  

```
2015-02-21_01:29:39.60887 {
2015-02-21_01:29:39.60888   "@timestamp": "2015-02-21T01:29:39.608487", 
2015-02-21_01:29:39.60889   "@version": 1, 
2015-02-21_01:29:39.60889   "host": "vm", 
2015-02-21_01:29:39.60889   "level": 3, 
2015-02-21_01:29:39.60890   "message": "Error when performing convergence: TypeError(\"'NoneType' object has no attribute '__getitem__'\",)", 
2015-02-21_01:29:39.60891   "otter_facility": "otter.rest.groups.delete_scaling_group", 
2015-02-21_01:29:39.60891   "otter_msg_type": "convergence-perform-error", 
2015-02-21_01:29:39.60891   "scaling_group_id": "7ec2111c-2145-46e8-a307-1fbaae429ddd", 
2015-02-21_01:29:39.60892   "tenant_id": "000000", 
2015-02-21_01:29:39.60892   "traceback": <snip>, 
2015-02-21_01:29:39.60897   "transaction_id": "bb4e8db6-f9d4-40d4-b988-5aa8510e0427"
2015-02-21_01:29:39.60897 }
```

And on this branch it looks like:

```
2015-02-21_01:22:40.17532 {
2015-02-21_01:22:40.17533   "@timestamp": "2015-02-21T01:22:40.175153", 
2015-02-21_01:22:40.17533   "@version": 1, 
2015-02-21_01:22:40.17533   "exception_type": "TypeError", 
2015-02-21_01:22:40.17534   "failure": "[Failure instance: <snip>\n]", 
2015-02-21_01:22:40.17534   "host": "vm", 
2015-02-21_01:22:40.17534   "level": 3, 
2015-02-21_01:22:40.17535   "message": "", 
2015-02-21_01:22:40.17535   "otter_facility": "otter.rest.groups.delete_scaling_group", 
2015-02-21_01:22:40.17535   "otter_msg_type": "convergence-perform-error", 
2015-02-21_01:22:40.17536   "scaling_group_id": "95d9a071-f989-4a71-889e-1e3f74ac9878", 
2015-02-21_01:22:40.17537   "tenant_id": "000000", 
2015-02-21_01:22:40.17537   "traceback": <snip>, 
2015-02-21_01:22:40.17538   "transaction_id": "5f7bfc8e-c8a2-4e01-98a4-2d69008288b7", 
2015-02-21_01:22:40.17539   "why": "Error when performing convergence"
2015-02-21_01:22:40.17539 }